test/apps/iodemo/io_demo.cc

```diff
@@ -172,48 +225,96 @@ class IoDemoRandom {
         }
     }
 
-    static inline void fill(unsigned &seed, void *buffer, size_t size) {
+    static inline void
+    fill(unsigned &seed, void *buffer, size_t size, bool force_host_memory)
+    {
         size_t body_count = size / sizeof(uint64_t);
         size_t tail_count = size & (sizeof(uint64_t) - 1);
         uint64_t *body    = reinterpret_cast<uint64_t*>(buffer);
         uint8_t *tail     = reinterpret_cast<uint8_t*>(body + body_count);
 
-        fill(seed, body, body_count);
-        fill(seed, tail, tail_count);
+        fill(seed, body, body_count, force_host_memory);
+        fill(seed, tail, tail_count, force_host_memory);
```

Comment:
since we agreed to fill/validate always on host in this PR, need to copy to/from device after/before host work is done in single `cudaMemcpy` call